docs/graphql/manual/schema/relay-schema.rst

```diff
@@ -0,0 +1,69 @@
+.. meta::
+   :description: Using Hasura's Relay API
+   :keywords: hasura, docs, Relay, schema, API
+
+.. _relay_schema:
+
+Relay schema
+============
+
+.. contents:: Table of contents
+  :backlinks: none
+  :depth: 1
+  :local:
+
+Introduction
+------------
+
+The Hasura GraphQL Engine serves a `Relay <https://relay.dev/>`__ schema for Postgres tables which have a primary key defined. The Relay schema can be accessed through the ``/v1/relay`` endpoint.
+
+.. thumbnail:: /img/graphql/manual/schema/relay.png
+   :alt: Relay API toggle
+
+What is Relay?
+--------------
+
+Relay is an opinionated JavaScript framework for declaratively fetching and managing GraphQL data. Relay's strength lies in how it removes opportunities for developer errors, by providing conventions that result in performant and type-safe apps.
+
+While using Relay saves you work on the client side with tasks like pagination, using Relay and Hasura together saves you even more work, because Hasura automatically sets up the Relay backend for you on top of your database.
+
+Benefits of Relay
+-----------------
+
+Relay's client-side benefits include:
+
+- **Colocation**: By convention, GraphQL fragments are colocated with their views, so that each component describes exactly what data it needs. This declarative approach has several benefits:
+
+  - It's hard to over-fetch data (which would hurt performance), or under-fetch data (which might cause errors).
+  - Components can only access data they've asked for. This **data masking** prevents implicit data dependency bugs.
+  - Components only re-render when the exact data they're using is updated, preventing unnecessary re-renders.
+
+- **Performance**: The Relay compiler composes your GraphQL fragments into optimized and efficient batches to reduce round-trips to the server. The compiler also applies **transforms** to your queries to remove redundancies and shorten query strings, which reduces upload bytes.
+
+- **Strong typing**: The compiler automatically generates Flow (or TypeScript) types, which you can import into your component for type checking in your code editor and during build time.
+
+To learn more about these and other Relay features, like persisted queries, local state management, passing arguments to fragments, and fetching data as early as possible, check out the `Relay docs <https://relay.dev/docs/en/experimental/a-guided-tour-of-relay>`__.
+
+.. note::
+  For a more detailed breakdown of Relay and its benefits, check out our `deep-dive on Relay <https://hasura.io/blog/deep-dive-into-relay-graphql-client/>`__.
+
+Relay's server spec
+-------------------
+
+To support the above client-side benefits, Relay has a particular server specification.
+
+*Hasura's Relay API sets up this spec for you automatically, so you don't have to implement it manually.*
+
+According to the spec, the server must provide:
```

Comment:
Not sure if this section is necessary since Hasura sets it up automatically, and users don't need to deal with this.

Reply:
would be great if you can show how to use the UPDATE_<table>_BY_PK funcitons that hasura generates for you.  I can't for the life of me figure out how to get that to work with a base64 encoded id returned by previous query.  I can't find example on this anywhere.  If you can explain how to work with base64 encoded ID's for update, query, (and update of FK to match PK in related table) I'll write it up for you afterwards.  Thanks!